kv/dist_sender.go

```diff
@@ -554,8 +554,10 @@ func (ds *DistSender) Send(call *client.Call) {
 					// On addressing errors, don't backoff; retry immediately.
 					return util.RetryReset, nil
 				case *proto.NotLeaderError:
-					ds.updateLeaderCache(proto.RaftID(desc.RaftID),
-						err.(*proto.NotLeaderError).GetLeader())
+					leader := err.(*proto.NotLeaderError).GetLeader()
```

Comment:
If you change the switch statement to `switch err := err.(type)` you can get rid of the type switch here.